### DISCUSSION TITLE
Script Discussion

### DISCUSSION THREAD

**Discussion by rahaaatul** [GuilleDiazFentanes/AutoClaim-AutoRoll-AutoReact-MudaeBot-2025#2] [General]
>> If I wasn't there, it'd be claimed by someone else, it didn't record Mikasa Ackerman, then another one. Because someone was doing bot commands. and it missed two of the cards. You need to do something about it, please @GuilleDiazFentanes

> I will think a form for the bot to do it, but if someone is spamming messages the message management that discord offers won’t be very solid to handle it

What's on your mind?

**Comment by rahaaatul** [comment]
![Screenshot (32)](https://github.com/GuilleDiazFentanes/AutoClaim-AutoRoll-AutoReact-MudaeBot-2023/assets/86149861/b0910540-ada9-46ab-8bb7-2eae3cbdaf76)

After I claimed Kakera, Mudae sent a _**Kakera Claimed**_ message, then the script ignored _**Crona**_  (probably because of the kakera message). And didn't attempt to claim it, even though it's in the list. As you know, I have added some code into it.

By the way, the bot doesn't try to click on that kakera, I have included the full list you gave me before.

```
desiredKakeras = ["kakera",'kakeraP','kakeraT','kakeraG','kakeraY','kakeraO','kakeraR','kakeraW','kakeraL']
```

**Reply by GuilleDiazFentanes** [reply]
yeah , its such a bad luck, if right on that moment the bot doesnt get the card info it will be impossible to claim it :/, If you donthave a lot of rolls i would put like a 2 second delay between rolls to decrease that case to happen

**Reply by GuilleDiazFentanes** [reply]
> By the way, the bot doesn't try to click on that kakera, I have included the full list you gave me before.


I thought they were all there, I dont know what name does that original kakera has, I recommend you to upgrade the gold badge as soon as you can, so that those kakeras always turn to yellow kakeras (450-550)

**Reply by rahaaatul** [reply]
> yeah , its such a bad luck, if right on that moment the bot doesnt get the card info it will be impossible to claim it :/, If you donthave a lot of rolls i would put like a 2 second delay between rolls to decrease that case to happen

Imagine, they think I stopped rolling because of the 2 sec delay and they start rolling and the script rolls after 2 sec delay. Both rolls at the same time 😅 Need to figure out other ways. 

> I thought they were all there, I dont know what name does that original kakera has, I recommend you to upgrade the gold badge as soon as you can, so that those kakeras always turn to yellow kakeras (450-550)

Yeah, I have already upgraded all my badges. Well, only on my main account. Need to find out what;s the name of it. How do i know the name of the emoji? Is it by chance `kakeraB` for `Blue Kakera`?

It is indeed `kakeraB`:
![image](https://github.com/GuilleDiazFentanes/AutoClaim-AutoRoll-AutoReact-MudaeBot-2024/assets/86149861/cfcb33ef-4b07-47a8-9c5e-4d02b5b99c19)

**Comment by GuilleDiazFentanes** [comment]
> What's on your mind?

The bot does something like this when it rolls:
 - Send message ( /wa)
 - Get last message sent into the channel( mudaes responde to /wa )
 - Get data from that last message sent
 - Do any action
 - Send message again (and repeat)
 
Taking into account that cicle, if someone puts other message in between it can contaminate the second step, being the "Get last message sent into the channel" a random text message, which will make the 3rd and 4th step impossible.

For the idea that you say, the cicle has to be changed to a "permanent" listening of ALL the messages of the channel ,so it will always get all the info to treat it.

**Reply by rahaaatul** [reply]
> For the idea that you say, the cicle has to be changed to a "permanent" listening of ALL the messages of the channel ,so it will always get all the info to treat it.

I'm all for it. This would be a better option. Keeps listening, matches wishlists, reacts immediately like a sniper. Sometimes my `account 2` have claim but the `account 1` don't. So if the `account 2` keeps listening, it can react quickly to the card that my `account 1` rolled but don't have the claim to claim. It's an important feature IMO.
Can you give me the code snippet for listening to all messages? I want to test it out.

**Comment by rahaaatul** [comment]
![Screenshot_2023_1230_230717](https://github.com/GuilleDiazFentanes/AutoClaim-AutoRoll-AutoReact-MudaeBot-2024/assets/86149861/4485771e-5f3f-4b53-8d13-1bc801ff6f6c)


For some reason, It didn't react to Nami 🫨!!!
She's in my wishlist in discord and in the text file. I don't understand, it didn't even record in the log. As you can see there's no message in between. And still misssed? 

Do you think it's because of the text above the embed?

**Reply by GuilleDiazFentanes** [reply]
Yes!! That embed text above and the heart emoji below makes it like it’s a “claimed” one. You need to change the server settings so that wishes don’t have a heart (I think it’s posible )

**Reply by rahaaatul** [reply]
Or we could set the emoji into kakera list? Like claiming the kakera

**Reply by rahaaatul** [reply]
If it treats it like it's claimed why did it not show in the log? Why does it completely ignored it? Do you a condition to add and treat it like it's unclaimed?

![Screenshot_2024_0102_125524](https://github.com/GuilleDiazFentanes/AutoClaim-AutoRoll-AutoReact-MudaeBot-2024/assets/86149861/8b66250b-ddaf-4d47-8b19-fe23ef3db30a)

I missed this one today, while I was sleeping.

**Reply by GuilleDiazFentanes** [reply]
I think that it is because of the content of the message, the normal rolls do not have text, that one has the “wished by …” 
Do you have the desires series or desired names? Maybe it does not recognize the serie

**Reply by GuilleDiazFentanes** [reply]
Try to set the wishes to private so it doesn’t get any text

**Reply by rahaaatul** [reply]
It doesn't even show up in the terminal, look at the [picture above](https://github.com/GuilleDiazFentanes/AutoClaim-AutoRoll-AutoReact-MudaeBot-2024/discussions/2#discussioncomment-7979807). It completely ignores "wished by" cards.

And regarding private wishes, even if i private it, someone else has a public wish for it. Take this as an example, My main account had Nico Robin as private wish and this alt account had it as public wish. It showed it anyway. Of course that account rolled it.

But the main thing here is, why is not updating it in terminal? Reacting to it comes later. Which I have a solution to and I wrote a code for it. Can you share your discord? I'd like to share it privately and get your opinion on it. Maybe you could implement it on your code in the future. It can click on any message that has wished by in it. but your code completely ignores it. 

Hoping for your reply asap! This is really an urgent matter.

**Reply by rahaaatul** [reply]
Hope this can help you, debug the issue:
This is the Nico Robin message:

```
{
  "id": "1191607500961755146",
  "type": 20,
  "content": "Wished by <@1234>",
  "channel_id": "1234",
  "author": {
    "id": "432610292342587392",
    "username": "Mudae",
    "avatar": "29cb28fbf65a3958105026ab03abd306",
    "discriminator": "0807",
    "public_flags": 65536,
    "premium_type": 0,
    "flags": 65536,
    "bot": true,
    "banner": null,
    "accent_color": null,
    "global_name": null,
    "avatar_decoration_data": null,
    "banner_color": null
  },
  "attachments": [],
  "embeds": [
    {
      "type": "rich",
      "description": "One Piece\n**927**<:kakera:469835869059153940>",
      "color": 1360437,
      "author": {
        "name": "Nico Robin"
      },
      "image": {
        "url": "https://imgur.com/Lq8Ctem.gif",
        "proxy_url": "https://images-ext-2.discordapp.net/external/D9h7AJLQ-J0tHsF4TXzypqyJxKvktjyJBFzqzU6W6ac/https/imgur.com/Lq8Ctem.gif",
        "width": 225,
        "height": 350
      },
      "footer": {
        "text": "Nico Robin / One Piece - 927 ka"
      }
    }
  ],
  "mentions": [
    {
      "id": "1234",
      "username": "name",
      "avatar": "avatar",
      "discriminator": "0",
      "public_flags": 0,
      "premium_type": 0,
      "flags": 0,
      "banner": null,
      "accent_color": null,
      "global_name": "name",
      "avatar_decoration_data": null,
      "banner_color": null
    }
  ],
  "mention_roles": [],
  "pinned": false,
  "mention_everyone": false,
  "tts": false,
  "timestamp": "2024-01-02T05:02:47.226000+00:00",
  "edited_timestamp": null,
  "flags": 0,
  "components": [
    {
      "type": 1,
      "components": [
        {
          "type": 2,
          "custom_id": "1191607500278079548p1059380001465696316",
          "style": 2,
          "emoji": {
            "name": "💓"
          }
        }
      ]
    }
  ],
  "application_id": "432610292342587392",
  "interaction": {
    "id": "1191607500278079548",
    "type": 2,
    "name": "wa",
    "user": {
      "id": "1234",
      "username": "name",
      "avatar": "avatar",
      "discriminator": "0",
      "public_flags": 0,
      "premium_type": 0,
      "flags": 0,
      "banner": null,
      "accent_color": null,
      "global_name": "name",
      "avatar_decoration_data": null,
      "banner_color": null
    }
  },
  "webhook_id": "1234",
  "resolved": {
    "users": {},
    "members": {},
    "channels": {},
    "roles": {}
  },
  "reactions": [
    {
      "emoji": {
        "id": null,
        "name": "🐿️"
      },
      "count": 1,
      "count_details": {
        "burst": 0,
        "normal": 1
      },
      "burst_colors": [],
      "me_burst": false,
      "burst_me": false,
      "me": true,
      "burst_count": 0
    }
  ]
}
```

**Reply by GuilleDiazFentanes** [reply]
Im on my phone but if I remember well it has to be in the very start of the while, look for the “continue” line of code, if the message has “content” it will skip the message as you say

**Reply by GuilleDiazFentanes** [reply]
Okeey, it is @cabezatabique

**Comment by kaanekr** [comment]
Advanced bot looks great. Does it already exist or is it being developed?

**Reply by Hyra666** [reply]
yeah ive been wondering too if it exists. Maybe we should wait until he releases it

**Reply by GuilleDiazFentanes** [reply]
im on it at the moment ^^

**Comment by Tegarcs142** [comment]
as i using this in my private server i noticed with too many keys in my server sometimes there will be 🛑 sign while rolling and could cause stopping the rolls so i modified a bit to prevent this issue. and that is cooldown between rolls
3k+ = 2 seconds
10k+ = 4 seconds

**Reply by RiassGremory** [reply]
can someone help me get this working it would be really appreciated none of the other scripts I've tried work

**Comment by RiassGremory** [comment]
anyone know how to make this work
